clients/cobol-lsp-vscode-extension/src/extension.ts

```diff
@@ -20,23 +20,27 @@ import { C4Z_FOLDER, GITIGNORE_FILE, LANGUAGE_ID } from "./constants";
 import { CopybookDownloadService } from "./services/copybook/CopybookDownloadService";
 import { CopybooksCodeActionProvider } from "./services/copybook/CopybooksCodeActionProvider";
 
+import { CommentAction, commentCommand } from "./commands/CommentCommand";
 import { initSmartTab } from "./commands/SmartTabCommand";
 import { LanguageClientService } from "./services/LanguageClientService";
 import { Middleware } from "./services/Middleware";
 import { TelemetryService } from "./services/reporter/TelemetryService";
 import { createFileWithGivenPath } from "./services/Settings";
 import { resolveSubroutineURI } from "./services/util/SubroutineUtils";
-import { CommentAction, commentCommand } from "./commands/CommentCommand";
+import { CompileTaskProvider } from "./task/CompileTaskProvider";
 
 let copyBooksDownloader: CopybookDownloadService;
 let middleware: Middleware;
 let languageClientService: LanguageClientService;
-
+let compileTaskProvider: vscode.Disposable | undefined;
+export const taskType = "cobolCompileJob";
 function initialize() {
     // We need lazy initialization to be able to mock this for unit testing
     copyBooksDownloader = new CopybookDownloadService();
     middleware = new Middleware(copyBooksDownloader);
     languageClientService = new LanguageClientService(middleware);
+    // create a task provider
+    compileTaskProvider = vscode.tasks.registerTaskProvider(taskType, new CompileTaskProvider());
 }
```

Comment:
Why define module level variables? Would it not be better if `initialize()` returned the newly create objects? For example

```js
function initialize() {
    const copyBooksDownloader = new CopybookDownloadService();
    const middleware = new Middleware(copyBooksDownloader);
    const languageClientService = new LanguageClientService(middleware);
    const compileTaskProvider = vscode.tasks.registerTaskProvider(taskType, new CompileTaskProvider());

    return {copyBooksDownloader, middleware, languageClientService, compileTaskProvider};
}
```

Reply:
updated